src/middleware/headers.js

```diff
@@ -60,4 +61,32 @@ module.exports = function (middleware) {
 
 		next();
 	};
+
+	let langs = [];
+	middleware.autoLocale = function (req, res, next) {
+		if (parseInt(req.uid, 10) > 0 || !meta.config.autoDetectLang) {
+			return next();
+		}
+
+		var lang = req.acceptsLanguages(langs);
+		if (!lang) {
+			return next();
+		}
+		req.query.lang = lang;
+		next();
+	};
+
+	languages.listCodes(function (err, codes) {
+		if (err) {
+			winston.error('[middleware/autoLocale] Could not retrieve languages codes list!');
+			codes = [];
+		}
+
+		winston.verbose('[middleware/autoLocale] Retrieves languages list for middleware');
+		var defaultLang = meta.config.defaultLang || 'en-GB';
+
+		langs = [defaultLang].concat(codes).filter(function (el, i, arr) {
+			return arr.indexOf(el) === i;
+		});
+	});
```

Comment:
This middleware is the same as the code removed below from `setupAutoLocale()`, in webserver.js